clang/lib/Sema/SemaChecking.cpp

```diff
@@ -13945,7 +13945,8 @@ void Sema::CheckArgumentWithTypeTag(const ArgumentWithTypeTagAttr *Attr,
 
 void Sema::AddPotentialMisalignedMembers(Expr *E, RecordDecl *RD, ValueDecl *MD,
                                          CharUnits Alignment) {
-  MisalignedMembers.emplace_back(E, RD, MD, Alignment);
+  if (!DisableSubstitionDiagnostics)
+    MisalignedMembers.emplace_back(E, RD, MD, Alignment);
```

Comment:
When this method is called from PruneTemporaryBindings (as a result of calls to CheckUnaryOperator), there should be no (duplicate) warning messages emitted about taking an address of a packed member. Removing this guard causes the Clang test Sema/address-unaligned.c to fail.